aws/table_aws_codestar_notification_rule.go

```diff
@@ -0,0 +1,188 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/codestarnotifications"
+
+	"github.com/turbot/go-kit/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsCodestarNotificationRule(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_codestar_notification_rule",
+		Description: "AWS CodeStar Notification Rule",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("arn"),
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"NotFound", "InvalidParameter"}),
+			},
+			Hydrate: describeNotificationRule,
+			Tags:    map[string]string{"service": "codestar-notifications", "action": "DescribeNotificationRule"},
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listNotificationRules,
+			Tags:    map[string]string{"service": "codestar-notifications", "action": "ListNotificationRules"},
+		},
+		HydrateConfig:     []plugin.HydrateConfig{},
+		GetMatrixItemFunc: SupportedRegionMatrix("codestar-notifications"), // unsure how to get EndpointsID from the package
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) of the notification rule.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Arn"),
+			},
+			{
+				Name:        "name",
+				Description: "The name of the notification rule.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     describeNotificationRule,
+				Transform:   transform.FromField("Name"),
```

Comment:
```suggestion
```
Transform function isn't needed since `FromField` transformation is applied by default 👍 !!

@gertvv you can remove all the references of `Transform:   transform.FromField` where by default the API returns data in upper camel case.